evalml/preprocessing/utils.py

```diff
@@ -50,10 +55,11 @@ def split_data(X, y, regression=False, test_size=.2, random_state=None):
     Returns:
         pd.DataFrame, pd.DataFrame, pd.Series, pd.Series: Feature and target data each split into train and test sets
     """
-    if not isinstance(X, pd.DataFrame):
-        X = pd.DataFrame(X)
-    if not isinstance(y, pd.Series):
-        y = pd.Series(y)
+    X = _convert_to_woodwork_structure(X)
+    y = _convert_to_woodwork_structure(y)
+
+    X = _convert_woodwork_types_wrapper(X.to_dataframe())
+    y = _convert_woodwork_types_wrapper(y.to_series())
 
     if regression:
```

Comment:
`ShuffleSplit` runs into the same errors when passed Woodwork, so converting to pandas is necessary :(

Reply:
@dsherry Ah right, we can use `iloc` now, so `train, test = next(CV_method.split(X.to_dataframe(), y.to_series()))` works. I think `index` works somewhat differently in Woodwork, since it was set to None for our test data 🤔